undertow/src/main/java/org/wildfly/extension/undertow/ServerAdd.java

```diff
@@ -92,7 +94,11 @@ protected void recordCapabilitiesAndRequirements(OperationContext context, Model
         // We currently don't have any attributes that reference capabilities, but we have this code in case that changes
         // since we are not calling the superclass code.
         ModelNode model = resource.getModel();
-        for (AttributeDefinition ad : getAttributes()) {
+
+        Map<String, AttributeAccess> attributeAccessMap = context.getResourceRegistration().getAttributes(PathAddress.EMPTY_ADDRESS);
+        for (Map.Entry<String, AttributeAccess> entry : attributeAccessMap.entrySet()) {
+            AttributeAccess attributeAccess = entry.getValue();
+            AttributeDefinition ad = attributeAccess.getAttributeDefinition();
```

Comment:
Technically, we should skip requirement recording for runtime attributes and aliases.
See: https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java#L259
I don't see any reason not to replace the first half of this method with a call to the super implementation, do you?

Reply:
@pferraro It looks like this method Java Doc responds to your question:

https://github.com/wildfly/wildfly/pull/18076/files#diff-9274d3139c44afa3e06390c846378a7a27d2b136ed92ac1050b193b86d0063e4R81-R83

Reply:
ok.